iosxr/l2_interfaces/iosxr_l2_interface.yaml

```diff
@@ -0,0 +1,93 @@
+---
+GENERATOR_VERSION: '1.0'
+
+ANSIBLE_METADATA: |
+    {
+      'metadata_version': '1.1',
+      'status': ['preview'],
+      'supported_by': 'network'
+    }
+NETWORK_OS: iosxr
+RESOURCE: l2_interfaces
+COPYRIGHT: Copyright 2019 Red Hat
+
+DOCUMENTATION: |
+  module: iosxr_l2_interfaces
+  version_added: 2.9
+  short_description: Manage Layer-2 interface on Cisco IOS-XR devices.
+  description: This module provides declarative management of Layer-2 interface on Cisco IOS-XR devices.
+  author: Sumit Jaiswal (@justjais)
+  options:
+    config:
+      description: A dictionary of Layer-2 interface options
+      type: list
+      elements: dict
+      suboptions:
+        name:
+          description:
+          - Full name of the interface excluding any logical unit number, i.e. GigabitEthernet0/0/0/1.
+          type: str
+          required: True
+        native_vlan:
+          description:
+          - Configure a native VLAN ID for the trunk
+          type: str
+        l2transport:
+          description:
+          - Switchport mode access command to configure the interface as a layer 2 access.
+          suboptions:
+            service_policy:
```

Comment:
Let's remove the service_policy, we can handle that with qos resource modules in the future

Reply:
and l2transport can be a bool to indicate the port in an AC